### PR TITLE
Add nimble formula

### PR DESCRIPTION
**Pull request by dambrisco**

### CODE REVIEW

File: Library/Formula/nimble.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "nimrod"
+
+  def install
+    system "nim", "c", "-r", "src/nimble", "install"
+    bin.install "nimble"
+  end
+
+  test do
+    print "No tests available at this time\n"
```
Comment (bfontaine) [inline Library/Formula/nimble.rb]: Please add a test here.

Comment (dambrisco) [inline Library/Formula/nimble.rb]: Added a couple binary tests. I'm not positive that nimble's own tester.nim suite actually passes (currently doesn't omm), so I've omitted that for the time being. Planning to reach out to the maintainer to figure that one out.

File: Library/Formula/nimble.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "nimrod"
+
+  def install
+    system "nim", "c", "-r", "src/nimble", "install"
+    bin.install "nimble"
+  end
+
+  test do
+    result = shell_output("#{bin}/nimble --version").chomp
+    assert result.start_with? "nimble"
```
Comment (DomT4) [inline Library/Formula/nimble.rb]: You can squash these two lines into `assert_match /nimble/, shell_output("#{bin}/nimble --version")`

File: Library/Formula/nimble.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    expected = "FAILURE: Please specify a search string."
+    actual = shell_output("#{bin}/nimble search", 1).chomp
+    assert_equal expected, actual
```
Comment (DomT4) [inline Library/Formula/nimble.rb]: And these into: `assert_match /Please specify a search string/, shell_output("#{bin}/nimble search", 1)`.

File: Library/Formula/nimble.rb (hunk 2 above)
Comment (bfontaine) [inline Library/Formula/nimble.rb]: @DomT4 it should be `/^nimble/` to match the  `start_with?`.

Comment (DomT4) [inline Library/Formula/nimble.rb]: :+1:

File: Library/Formula/nimble.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  depends_on "nimrod"
+
+  def install
+    system "nim", "c", "-r", "src/nimble", "install"
+    bin.install "nimble"
+  end
+
+  test do
+    assert_match /^nimble/, shell_output("#{bin}/nimble --version")
+    assert_match /Please specify a search string/, shell_output("#{bin}/nimble search", 1)
```
Comment (mikemcquaid) [inline Library/Formula/nimble.rb]: Could you install a package?

Comment (DomT4) [inline Library/Formula/nimble.rb]: Ping on Mike's comment. Would be great if we can install something to the `testpath`.

**Comment by DomT4** [comment]
Closing as dead. Feel free to address the comments resubmit this fresh if still interested. Thanks for the submission!

**Pull request closed without merging**